crypto/signer.go

```diff
@@ -30,26 +31,23 @@ var sigpbHashLookup = map[crypto.Hash]sigpb.DigitallySigned_HashAlgorithm{
 // Signer is responsible for signing log-related data and producing the appropriate
 // application specific signature objects.
 type Signer struct {
-	hash         crypto.Hash
-	signer       crypto.Signer
-	sigAlgorithm sigpb.DigitallySigned_SignatureAlgorithm
+	hash   crypto.Hash
+	signer crypto.Signer
 }
 
 // NewSigner creates a new Signer wrapping up a hasher and a signer. For the moment
 // we only support SHA256 hashing and either ECDSA or RSA signing but this is not enforced
 // here.
-func NewSigner(sigAlgo sigpb.DigitallySigned_SignatureAlgorithm, signer crypto.Signer) *Signer {
+func NewSigner(signer crypto.Signer) *Signer {
 	return &Signer{
-		hash:         crypto.SHA256,
-		signer:       signer,
-		sigAlgorithm: sigAlgo,
+		hash:   crypto.SHA256,
+		signer: signer,
 	}
 }
 
-// NewSignerFromPrivateKeyManager creates a new Signer wrapping up a hasher and a private key.
-// For the moment, we only support SHA256 hashing and either ECDSA or RSA signing but this is not enforced here.
-func NewSignerFromPrivateKeyManager(key PrivateKeyManager) *Signer {
-	return NewSigner(key.SignatureAlgorithm(), key)
+// Public returns the public key that can verify signatures produced by s.
+func (s *Signer) Public() crypto.PublicKey {
```

Comment:
There are two reasons to have this method:
1) It keeps the methods exposed by `Signer` as close to the `crypto.Signer` interface as possible.
2) By exposing the `crypto.PublicKey`, the `keys.SignatureAlgorithm()` func can be used to determine the `sigpb.DigitallySigned_SignatureAlgorithm` used by the `Signer`.